File: 941-sort-array-by-parity/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        evenPos=0
        i=0
        n = len(nums)
        
        while i<n:
            if not nums[i]&1:
                nums[i],nums[evenPos] = nums[evenPos], nums[i]
                evenPos+=1
            i+=1
        
        return nums

File: 941-sort-array-by-parity/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int evenPos = 0;
        for(int i=0;i<nums.size();i++){
            if(~nums[i]&1){
                swap(nums[i],nums[evenPos]);
                evenPos+=1;
            }
        }
        return nums;
    }
};